provisioning/transport/mqtt/src/ProvisioningChannelHandlerAdapter.cs

```diff
@@ -326,6 +333,29 @@ private async Task VerifyPublishPacketTopicAsync(IChannelHandlerContext context,
             }
         }
 
+        private static TimeSpan? getRetryAfterFromTopic(string topic)
+        {
+            string[] topicAndQueryString = topic.Split('?');
+            if (topicAndQueryString.Length > 1)
+            {
+                string[] queryPairs = topicAndQueryString[1].Split('&');
+                for (int queryPairIndex = 0; queryPairIndex < queryPairs.Length; queryPairIndex++)
+                {
+                    string[] queryKeyAndValue = queryPairs[queryPairIndex].Split('=');
+                    if (queryKeyAndValue.Length == 2 && queryKeyAndValue[0].Equals(RetryAfter, StringComparison.OrdinalIgnoreCase))
+                    {
+                        int secondsToWait;
+                        if (int.TryParse(queryKeyAndValue[1], out secondsToWait))
+                        {
+                            return TimeSpan.FromSeconds(secondsToWait);
+                        }
+                    }
+                }
+            }
+
+            return null;
```

Comment:
I think we should throw if invalid data is sent from the service and crash the application.
(If the topic _does_ have the data then it _must_ be correct. Otherwise we can ignore.) /cc @alextolp

Reply:
I don't see a reason to drop the whole connection simply because the retry-after field is off. If we just fall back to the default retry operation, the SDK will handle this scenario just fine.